src/App.jsx

```diff
@@ -14,45 +17,107 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_BY_ALPHABET = 'alphabet';
+const SORT_BY_LENGTH = 'length';
+const RESET = 'reset';
+
+function getPreparedGoods(goods, sortFilter, reverseOrder) {
+  const sortedGoods = [...goods];
+
+  sortedGoods.sort((good1, good2) => {
+    switch (sortFilter) {
+      case SORT_BY_ALPHABET: {
+        return good1.localeCompare(good2);
+      }
+
+      case SORT_BY_LENGTH:
+        return good1.length - good2.length;
+
+      default:
+        return 0;
+    }
+  });
+
+  return !reverseOrder ? sortedGoods : sortedGoods.reverse();
+}
+
+export const App = () => {
+  const [sortFilter, setSortFilter] = useState('');
+  const [reverseOrder, setReverseOrder] = useState(false);
+  const visibleGoods = getPreparedGoods(
+    goodsFromServer,
+    sortFilter,
+    reverseOrder,
+  );
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  function toggleReverse() {
+    setReverseOrder(!reverseOrder);
+  }
+
+  function handleReset() {
+    setSortFilter(RESET);
+    setReverseOrder(false);
+  }
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={cn({
+            'is-light': sortFilter !== SORT_BY_ALPHABET,
+          }, 'button is-info')}
+          onClick={() => setSortFilter(SORT_BY_ALPHABET)}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          className={cn({
+            'is-light': sortFilter !== SORT_BY_LENGTH,
+          }, 'button is-success')}
+          onClick={() => setSortFilter(SORT_BY_LENGTH)}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          className={cn({
+            'is-light': !reverseOrder,
+          }, 'button is-warning')}
+          onClick={toggleReverse}
+        >
+          Reverse
+        </button>
+
+        {JSON.stringify(visibleGoods) === JSON.stringify(goodsFromServer)
```

Comment:
Show reset button if isReverseOrder is true or sortFilter is not equal to empty string